src/components/HomePage.tsx

```diff
@@ -193,14 +160,18 @@ export default function HomePage(props: Props) {
         <SessionInfo session={session} />
       ) : (
         <AuthForm
-          setupSession={setupSession}
-          serverChange={serverChange}
-          getServerInfo={getServerInfo}
+          setupSession={auth => dispatch(SessionActions.setupSession(auth))}
+          serverChange={() => dispatch(SessionActions.serverChange())}
+          getServerInfo={auth => dispatch(SessionActions.getServerInfo(auth))}
           session={session}
           servers={servers}
-          clearServers={clearServers}
-          navigateToExternalAuth={navigateToExternalAuth}
-          navigateToOpenID={navigateToOpenID}
+          clearServers={() => dispatch(ServersActions.clearServers())}
+          navigateToExternalAuth={authFormData =>
+            dispatch(SessionActions.navigateToExternalAuth(authFormData))
+          }
+          navigateToOpenID={(authFormData, provider) =>
+            dispatch(SessionActions.navigateToOpenID(authFormData, provider))
+          }
```

Comment:
This is pretty ugly, but temporary. Eventually, we'll be able to remove many of these props and access this redux state / dispatch actions directly from `AuthForm`.